storm-client/src/jvm/org/apache/storm/Config.java

```diff
@@ -1884,4 +1888,11 @@ public void setTopologyStrategy(String strategy) {
         this.put(Config.TOPOLOGY_SCHEDULER_STRATEGY, strategy);
     }
 
+    public static String getBlobstoreHDFSPrincipal(Map conf) throws UnknownHostException {
+        String principal = (String)conf.get(Config.BLOBSTORE_HDFS_PRINCIPAL);
+        if (principal != null) {
+            principal = principal.replace("HOSTNAME", Utils.localHostname());
```

Comment:
Hadoop picks `_HOST` as placeholder and it would have less change to be conflicted. So if we have no better reason to use alternative, using `_HOST` would be better to me.

Also I would feel more safer if we could try to parse principal and only replace placeholder when host part in principal is just a placeholder.  So `_HOST` in `yarn/_HOST@EXAMPLE.COM` can be substituted, but both `_HOST` in `yarn/_HOST2@EXAMPLE.COM` and `_HOST` in `_HOST/host1@EXAMPLE.COM` should not be substituted.

@revans2 Could you provide some suggestions?

Reply:
I agree _HOST is probably preferable, as it is less likely to result in a collision by accident, and it follows with what Hadoop is doing.